java/code/src/com/redhat/rhn/taskomatic/task/ForwardRegistrationTask.java

```diff
@@ -0,0 +1,107 @@
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.rhn.taskomatic.task;
+
+import com.redhat.rhn.domain.credentials.Credentials;
+import com.redhat.rhn.domain.product.SUSEProduct;
+import com.redhat.rhn.domain.scc.SCCCachingFactory;
+import com.redhat.rhn.domain.scc.SCCDeregisterItem;
+import com.redhat.rhn.domain.server.Server;
+import com.redhat.rhn.domain.server.ServerFactory;
+
+import com.suse.scc.model.SCCMinProductJson;
+import com.suse.scc.model.SCCRegisterSystemJson;
+import com.suse.utils.Opt;
+
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+public class ForwardRegistrationTask extends RhnJavaJob {
+
+    @Override
+    public void execute(JobExecutionContext arg0) throws JobExecutionException {
+        List<Server> forwardRegistration = ServerFactory.findSystemsToForwardRegistration();
+        List<SCCDeregisterItem> deregister = SCCCachingFactory.listDeregisterItems();
+
+
+    }
+
+    public SCCRegisterSystemJson getPayload(Credentials creds, Server srv) {
+        List<SCCMinProductJson> products = Opt.fold(srv.getInstalledProductSet(),
+                () -> {
+                    return new LinkedList<SUSEProduct>();
+                },
+                s -> {
+                    List<SUSEProduct> prd = new LinkedList<>();
+                    prd.add(s.getBaseProduct());
+                    prd.addAll(s.getAddonProducts());
+                    return prd;
+                }
+                ).stream()
+        .map(p -> new SCCMinProductJson(p))
+        .collect(Collectors.toList());
+
+        Map<String, String> hwinfo = new HashMap<>();
+        Optional.ofNullable(srv.getCpu().getNrCPU()).ifPresent(c -> hwinfo.put("cpus", c.toString()));
+        Optional.ofNullable(srv.getCpu().getNrsocket()).ifPresent(c -> hwinfo.put("sockets", c.toString()));
+        hwinfo.put("arch", srv.getServerArch().getLabel().split("-")[0]);
+        if (srv.isVirtualGuest()) {
+            String cloud = "";
+            String hv = srv.getVirtualInstance().getType().getLabel();
+            switch(hv) {
+            case "fully_virtualized":
+            case "para_virtualized":
+                hv = "XEN";
+                break;
+            case "qemu":
+                hv = "KVM";
+                break;
+            case "vmware":
+            case "hyperv":
+                break;
+            case "azure":
+                cloud = "Microsoft";
+                break;
+            case "aws":
+                cloud = "Amazon";
+                break;
+            case "gce":
+                cloud = "Google";
+                break;
+            case "nutanix":
+                cloud = "Nutanix";
```

Comment:
Nutanix is not a cloud but a hypervisor, like VMware or Hyper-V

Reply:
changed